### PR TITLE
Gracefully exit when extra dependencies are missing

### PR DESCRIPTION
**Pull request by charlesbvll**
#### What does this implement/fix? Explain your changes.

This PR makes Flower exit gracefully when people are trying to run code using REST without having installed the dependencies.

### CODE REVIEW

**Comment by charlesbvll** [comment]
This is before:

image

This is after:

image

File: src/py/flwr/common/logger.py [hunk 1]
```diff
@@ -13,6 +13,8 @@
 # limitations under the License.
 # ==============================================================================
 """Flower Logger."""
+
+
```
Comment (charlesbvll) [inline src/py/flwr/common/logger.py:17]:
```suggestion
```

File: src/py/flwr/common/telemetry.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
 # ==============================================================================
 """Flower telemetry."""
 
+
```
Comment (charlesbvll) [inline src/py/flwr/common/telemetry.py:17]:
```suggestion
```

File: src/py/flwr/common/version.py [hunk 3]
```diff
@@ -1,4 +1,6 @@
 """Flower package version helper."""
+
+
```
Comment (charlesbvll) [inline src/py/flwr/common/version.py:3]:
```suggestion
```

File: src/py/flwr/common/telemetry_test.py [hunk 4]
```diff
@@ -14,6 +14,7 @@
 # ==============================================================================
 """Telemetry tests."""
 
+
```
Comment (charlesbvll) [inline src/py/flwr/common/telemetry_test.py:17]:
```suggestion

```

**Review by tanertopal — DISMISSED** [review]

File: src/py/flwr/common/constant.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+# ==============================================================================
+"""Flower constants."""
+
+
+MISSING_EXTRA_REST = """
+Extra dependencies required for using the REST-based Fleet API are missing.
+
+To use the REST API, install `flwr` with the `rest` extra:
+
+    `pip install flwr['rest']`.
```
Comment (danieljanes) [inline src/py/flwr/common/constant.py:23]:
```suggestion
    `pip install flwr[rest]`.
```

**Pull request merged**